### PR TITLE
executor: remove Next function for executor/write.go

### PR DESCRIPTION
**Pull request by yangwenmai**

### CODE REVIEW

**Comment by zz-jason** [comment]
@yangwenmai please remove the unused function and struct fields as well:
```sh
➜ unused github.com/pingcap/tidb/executor
executor/executor.go:898:2: field asName is unused (U1000)
executor/show.go:61:2: field fetched is unused (U1000)
executor/write.go:212:22: func (*DeleteExec).deleteMultiTables is unused (U1000)
executor/write.go:249:22: func (*DeleteExec).deleteSingleTable is unused (U1000)
executor/write.go:1295:24: func (*InsertValues).getRowsSelect is unused (U1000)
executor/write.go:1811:22: func (*UpdateExec).fetchRows is unused (U1000)
```

**Comment by zz-jason** [comment]
to #5985

File: executor/write.go [hunk 1]
```diff
@@ -812,11 +734,6 @@ func (e *LoadData) exec(ctx context.Context) (Row, error) {
 	return nil, nil
 }
 
-// Next implements the Executor Next interface.
-func (e *LoadData) Next(ctx context.Context) (Row, error) {
-	return e.exec(ctx)
-}
-
 // NextChunk implements the Executor NextChunk interface.
 func (e *LoadData) NextChunk(ctx context.Context, chk *chunk.Chunk) error {
```
Comment (zz-jason) [inline executor/write.go]: we can remove `exec` and put the logical inside `NextChunk`

Comment (yangwenmai) [reply executor/write.go]:
```go
    if e.done {
        return nil
    }
    err := e.run(ctx)
    e.done = true
    return errors.Trace(err)
```
Is it?

Comment (zz-jason) [reply executor/write.go]: no, I mean we can remove function `exec` and put the codes of `exec` into `NextChunk`

Comment (yangwenmai) [reply executor/write.go]: I mean I remove Next() and put that codes of `exec` into `NextChunk`.

Comment (yangwenmai) [reply executor/write.go]: Only LoadData? Because it makes the code simple.

**Comment by yangwenmai** [comment]
ci error 1:

```
fatal error: runtime: out of memory
```

ci error 2:

```
gofmt (simplify)
./executor/executor.go
make: *** [check] Error 1
```

I don't know what is wrong.

**Comment by zz-jason** [comment]
@yangwenmai you can run `make check` to fix ci

**Comment by shenli** [comment]
LGTM

**Pull request merged**